setup-gcloud/src/version-util.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Contains version utility functions.
+ */
+
+import * as httpm from 'typed-rest-client/HttpClient';
+
+/**
+ * @returns The latest stable version of the gcloud SDK.
+ */
+export async function getLatestGcloudSDKVersion(): Promise<string> {
+  const client: httpm.HttpClient = new httpm.HttpClient(
+    'github-actions-setup-gcloud',
+  );
+  return client
+    .get(
+      'https://api.github.com/repos/GoogleCloudPlatform/cloud-sdk-docker/tags',
+    )
+    .then(res => {
```

Comment:
Can we change this to async/await pattern to stay consistent?

Reply:
I would love to, sadly the library we're using here doesn't support it, so we have to wrap around it :)

Reply:
I'm seeing async/await in their "docs" aka tests (https://github.com/microsoft/typed-rest-client/blob/master/test/tests/httptests.ts#L52). This code is just getting harder and harder to read.

Reply:
Oh snap! You are correct. For some reason when I first looked into using the library I had decided it didn't support it. Glad to see that's not the case. Good catch!